### PR TITLE
General consolidation / updates

### PR DESCRIPTION
**Pull request by liamjohnston**
- consolidate django/silverstripe files. 
- update all the things
- remove some things that are probably outdated
- add some things that were missing (e.g. .prettierrc, .browserslistrc)

**Probably todo (separate PR):** remove the wagtail and silverstripe directories as this supersedes them

### CODE REVIEW

File: webpack-build/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "trailingComma": "all",
+  "tabWidth": 2,
+  "semi": true,
+  "singleQuote": true,
+  "arrowParens": "always"
+}
```
Comment (liamjohnston) [inline webpack-build/.prettierrc:7]: My opinions :D

File: webpack-build/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+> **Setting project-wide yarn version**
+> You may like to set the yarn version for the entire project to avoid people from using different versions locally. To do that run `yarn policies set-version` (see [yarn policies](https://yarnpkg.com/lang/en/docs/cli/policies/)).
+
+```zsh
+nvm install
+nvm use
+yarn policies set-version
+yarn install
+```
```
Comment (liamjohnston) [inline webpack-build/README.md:46]: Does this bit belong in here?

File: webpack-build/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- [`custom.d.ts`](./custom.d.ts)
+
+### Step 2
+
+Update things as needed. Specifically:
+
+- In `package.json`, change the 'cms' variable. If you're using Silverstripe, set it to `silverstripe`. If you're using Django/wagtail, set it to `django`. If you're using something else (or no CMS), you will need to update `webpack.congig.json`. If you do, **please make a PR** that adds this new setup as an option for other future projects.
+- Update `.browserslistrc` if necessary. Currently it is evergreen only (modern browsers, i.e. no IE).
+- Change any other stuff that you want to change. This isn't a list of rules, just a startring point. If you make a change you think is a general improvement that will help others, **please make a PR**.
+
```
Comment (liamjohnston) [inline webpack-build/README.md:35]: Any other things we should recommend people check or change?

File: webpack-build/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+### webpack.config.js and webpack.fixpaths.js
+
+The main Webpack configuration, supporting Babel and Sass. Our config works around some different assumptions between Webpack and DW relating to project file structure and webserver URLs. Specifically DW has templates in one directory and static files in another (`core/templates` and `core/static`) but the webserver URLs for these directories aren't side-by-side, so the `webpack.fixpaths.js` fixes these paths.
+
+If you need to make DW use different file structures or webserver URLs then this is the place to configure it.
+
+### .prettierignore
+
+Prettier seems to mangle our Django templates because they have template `{% if %}` syntax inside `.html` files but they're not HTML. This `.prettierignore` file ensures that `.html` won't be reformatted for this reason.
```
Comment (liamjohnston) [inline webpack-build/README.md:67]: Not sure if we should update this further or remove...

**Comment by liamjohnston** [comment]
@holloway Please let me know your thoughts. Aside from the comments in the PR, some further thoughts:
- Should that base.html be even more paired down? It needed to exist for the current webpack thing to run but maybe should only contain an example of the react component thing?
- Should we completely remove the wagtail/silverstripe directories?? Seems like this supersedes them, assuming the HTML structure is outside of the scope (remembering that this project is essentially just a list of files they can copy over, not a basis for a new project).

**Comment by holloway** [comment]
@liamjohnston nice work!

Because this is adding a directory, I'm happy to merge it and address any issues in subsequent PRs :)

yep, let's pair it (base.html) down a bit more if possible.

File: webpack-build/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- [`.eslintignore`](./.eslintignore)
+- [`.babelrc`](./.babelrc)
+- [`tsconfig.json`](./tsconfig.json)
+- [`custom.d.ts`](./custom.d.ts)
+
+### Step 2
+
+Update things as needed. Specifically:
+
+- In `package.json`, change the 'cms' variable. If you're using Silverstripe, set it to `silverstripe`. If you're using Django/wagtail, set it to `django`. If you're using something else (or no CMS), you will need to update `webpack.congig.json`. If you do, **please make a PR** that adds this new setup as an option for other future projects.
```
Comment (holloway) [inline webpack-build/README.md:32]: congig.json

File: webpack-build/README.md (hunk 4 above)
Comment (holloway) [reply webpack-build/README.md:68]: Nah this is good

File: webpack-build/README.md (hunk 2 above)
Comment (holloway) [reply webpack-build/README.md:46]: Yep I think that's good.

File: webpack-build/.prettierrc (hunk 1 above)
Comment (holloway) [reply webpack-build/.prettierrc:7]: I agree

File: webpack-build/package.json [hunk 6]
```diff
@@ -0,0 +1,63 @@
+{
+  "cms": "silverstripe",
+  "name": "YOUR_PROJECT",
+  "version": "1.0.0",
+  "private": true,
+  "description": "YOUR_PROJECT - built with Silverstripe",
+  "main": "index.js",
```
Comment (holloway) [inline webpack-build/package.json:7]: Delete

File: webpack-build/package.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
+{
+  "cms": "silverstripe",
+  "name": "YOUR_PROJECT",
+  "version": "1.0.0",
+  "private": true,
+  "description": "YOUR_PROJECT - built with Silverstripe",
```
Comment (holloway) [inline webpack-build/package.json:6]: Probably shouldn't mention silverstripe

File: webpack-build/src/index.tsx [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+//polyfills?
+
+//some component
```
Comment (holloway) [inline webpack-build/src/index.tsx:5]: Should this import `./components.tsx` ?

We probably don't need to import React from here

**Comment by holloway** [comment]
A long time ago we used to have a `recycle-bin` directory where we dumped files that are deprecated and would be removed in the future.

https://github.com/springload/frontend-starter-kit/tree/c35abe435eb4905565e79a8bbc5cc257b2e5a77c

I suggest we do the same to the SilverStripe and Django directories (perhaps in another PR)

**Comment by liamjohnston** [comment]
> A long time ago we used to have a `recycle-bin` directory where we dumped files that are deprecated and would be removed in the future.
> 
> https://github.com/springload/frontend-starter-kit/tree/c35abe435eb4905565e79a8bbc5cc257b2e5a77c
> 
> I suggest we do the same to the SilverStripe and Django directories (perhaps in another PR)

Huh, ok... seems a little weird to me but I'll do it :)

**Pull request merged**